torch/functional.py

```diff
@@ -695,80 +696,74 @@ def stft(input: Tensor, n_fft: int, hop_length: Optional[int] = None,
-
-    The :attr:`n_fft`, :attr:`hop_length`, :attr:`win_length` are all the same which prevents the calculation
-    of right padding. These additional values could be zeros or a reflection of the signal so providing
-    :attr:`length` could be useful. If :attr:`length` is ``None`` then padding will be aggressively removed
-    (some loss of signal).
-
-    [1] D. W. Griffin and J. S. Lim, "Signal estimation from modified short-time Fourier transform,"
-    IEEE Trans. ASSP, vol.32, no.2, pp.236-243, Apr. 1984.
-
-    Args:
-        input (Tensor): The input tensor. Expected to be output of :func:`~torch.stft`,
-            can either be complex (``channel``, ``fft_size``, ``n_frame``), or real
-            (``channel``, ``fft_size``, ``n_frame``, 2) where the ``channel``
-            dimension is optional.
-
-            .. deprecated:: 1.8.0
-               Real input is deprecated, use complex inputs as returned by
-               ``stft(..., return_complex=True)`` instead.
-        n_fft (int): Size of Fourier transform
-        hop_length (Optional[int]): The distance between neighboring sliding window frames.
-            (Default: ``n_fft // 4``)
-        win_length (Optional[int]): The size of window frame and STFT filter. (Default: ``n_fft``)
-        window (Optional[torch.Tensor]): The optional window function.
-            (Default: ``torch.ones(win_length)``)
-        center (bool): Whether :attr:`input` was padded on both sides so that the :math:`t`-th frame is
-            centered at time :math:`t \times \text{hop\_length}`.
-            (Default: ``True``)
-        normalized (bool): Whether the STFT was normalized. (Default: ``False``)
-        onesided (Optional[bool]): Whether the STFT was onesided.
-            (Default: ``True`` if ``n_fft != fft_size`` in the input size)
-        length (Optional[int]): The amount to trim the signal by (i.e. the
-            original signal length). (Default: whole signal)
-        return_complex (Optional[bool]):
-            Whether the output should be complex, or if the input should be
-            assumed to derive from a real signal and window.
-            Note that this is incompatible with ``onesided=True``.
-            (Default: ``False``)
-
-    Returns:
-        Tensor: Least squares estimation of the original signal of size (..., signal_length)
-    """
-    if has_torch_function_unary(input):
-        return handle_torch_function(
-            istft, (input,), input, n_fft, hop_length=hop_length, win_length=win_length,
-            window=window, center=center, normalized=normalized, onesided=onesided,
-            length=length, return_complex=return_complex)
-
-    return _VF.istft(input, n_fft, hop_length, win_length, window, center,  # type: ignore[attr-defined]
-                     normalized, onesided, length, return_complex)
+istft = _add_docstr(
+    torch.istft,
+    "istft(input, n_fft, hop_length=None, win_length=None, window=None, center=True, "
```

Comment:
Actually, the c++ function calls the first argument "self" instead of input.
This means that calling `torch.functional.istft(input=foo, ...)` used to work but won't anymore after this change.
That being said, today, you have to do `torch.istft(self=, ...)` if you use the torch. version...

Note that torchaudio does call the function with input= here: https://github.com/pytorch/audio/blob/ed6256a26aef254ecd048d795dc5814a1f570851/torchaudio/functional/functional.py#L177

Not sure what to do here as there might be uses for the torch. version with self= as well in the wild...

Can you remove that change from this PR for now so that we can land the PR and discuss this more problematic change in a different one?

Reply:
The `self` argument in `native_functions.yaml` is renamed to `input` for the functional variants. You can see this in `python_torch_functionsEverything.cpp`:
```cpp
// istft
static PyObject * THPVariable_istft(PyObject* self_, PyObject* args, PyObject* kwargs)
{
  HANDLE_TH_ERRORS
  static PythonArgParser parser({
    "istft(Tensor input, int64_t n_fft, int64_t? hop_length=None, int64_t? win_length=None, Tensor? window=None, bool center=True, bool normalized=False, bool? onesided=None, int64_t? length=None, bool return_complex=False)",
  }, /*traceable=*/true);

  // ...
}
```
and if I try to pass `self=foo` then it fails:
```
----> 1 torch.functional.istft(self=torch.ones(()), n_fft=2)
TypeError: istft() missing 2 required positional argument: "input", "n_fft"
```

Reply:
But torchscript doesn't know about that :( 
So when torchaudio tries to torchscript their model it blows up:
```
Argument self not provided.
:
  File "...../pytorch/audio/test/torchaudio_unittest/transforms_torchscript_consistency_cpu_test#binary,link-tree/torchaudio/functional/functional.py", line 176

    # default values are consistent with librosa.core.spectrum._spectrogram
    waveform = torch.istft(
               ~~~~~~~~~~~ <--- HERE
        input=spectrogram,
        n_fft=n_fft,
'inverse_spectrogram' is being compiled since it was called from 'InverseSpectrogram.forward'
  File "/data/sandcastle/boxes/eden-trunk-hg-fbcode-fbsource/fbcode/buck-out/dev/gen/pytorch/audio/test/torchaudio_unittest/transforms_torchscript_consistency_cpu_test#binary,link-tree/torchaudio/transforms.py", line 205
            Tensor: Dimension (..., time), Least squares estimation of the original signal.
        """
        return F.inverse_spectrogram(
               ~~~~~~~~~~~~~~~~~~~~~~
            spectrogram,
            ~~~~~~~~~~~~
            length,
            ~~~~~~~
            self.pad,
            ~~~~~~~~~
            self.window,
            ~~~~~~~~~~~~
            self.n_fft,
            ~~~~~~~~~~~
            self.hop_length,
            ~~~~~~~~~~~~~~~~
            self.win_length,
            ~~~~~~~~~~~~~~~~
            self.normalized,
            ~~~~~~~~~~~~~~~~
            self.center,
            ~~~~~~~~~~~~
            self.pad_mode,
            ~~~~~~~~~~~~~~
            self.onesided,
            ~~~~~~~~~~~~~ <--- HERE
        )

```

Reply:
cc @eellison is there any plan to be able to fix these? At the moment since script expects self= and the python binding expects input= we cannot use kwargs here?

Reply:
Okay, I've removed the istft changes from this PR and opened an issue for the TorchScript bug #71994

Reply:
Awesome! Thanks!